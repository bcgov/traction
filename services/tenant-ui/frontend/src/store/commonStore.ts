/**
 * Place for storing common state that is not persisted in the backend.
 */
import { defineStore } from 'pinia';
import { ref } from 'vue';

export const useCommonStore = defineStore('commonState', () => {
  // state
  const sidebarOpen: any = ref(null);

  // return { sidebarOpen, setSidebarOpen };
  return { sidebarOpen };
});
