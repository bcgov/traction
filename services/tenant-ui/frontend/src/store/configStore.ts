import { defineStore } from 'pinia';
import { ref } from 'vue';
import axios from 'axios';

export const useConfigStore = defineStore('config', () => {
  // state
  const config: any = ref(null);
  const loading: any = ref(false);
  const error: any = ref(null);

  // getters
  function proxyPath(p: string) {
    if (config.value) {
      let pp = config.value.frontend.proxyPath;
      if (pp.endsWith('/')) {
        pp = pp.slice(0, -1);
      }
      if (!p.startsWith('/')) {
        p = `/${p}`;
      }

      return `${pp}${p}`;
    } else {
      // ?? error ??
      return p;
    }
  }

  // actions
  async function load() {
    console.log('> configStore.load');
    config.value = null;
    error.value = null;
    loading.value = true;
    await axios
      .get('config')
      .then((res) => {
        console.log(res);
        config.value = res.data;
      })
      .catch((err) => {
        error.value = err;
        //console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< configStore.load');
    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return config.value;
  }

  return { config, loading, error, load, proxyPath };
});

export default {
  useConfigStore,
};
