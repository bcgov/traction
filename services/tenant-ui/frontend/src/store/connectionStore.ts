import { API_PATH, CONNECTION_STATUSES } from '@/helpers/constants';
import {
  ConnRecord,
  UpdateConnectionRequest,
} from '@/types/acapyApi/acapyInterface';
import { defineStore } from 'pinia';
import { computed, ref, Ref } from 'vue';
import { useAcapyApi } from './acapyApi';
import {
  fetchList,
  filterByStateActive,
  filterMapSortList,
  sortByLabelAscending,
} from './utils';
import { fetchItem } from './utils/fetchItem';

export const useConnectionStore = defineStore('connection', () => {
  // state
  const connections: Ref<ConnRecord[]> = ref([]);
  const selectedConnection: any = ref(null);
  const loading: Ref<boolean> = ref(false);
  const loadingItem: Ref<boolean> = ref(false);
  const error: Ref<string | null> = ref(null);

  // getters
  const connectionsDropdown = computed(() => {
    // Get the display list of active connections from the util
    return filterMapSortList(
      connections.value,
      connectionLabelValue,
      sortByLabelAscending,
      filterByStateActive
    );
  });

  const filteredConnections = computed(() =>
    connections.value.filter((c) => c.state !== CONNECTION_STATUSES.INVITATION)
  );
  const filteredInvitations = computed(() =>
    connections.value.filter((c) => c.state === CONNECTION_STATUSES.INVITATION)
  );

  const findConnectionName = computed(() => (connectionId: string) => {
    if (loading.value) return undefined;
    // Find the connection alias for an ID
    const connection = connections.value?.find((c: any) => {
      return c.connection_id === connectionId;
    });
    return connection && connection.alias ? connection.alias : '';
  });

  // actions

  // grab the tenant api
  const acapyApi = useAcapyApi();

  async function listConnections() {
    selectedConnection.value = null;
    return fetchList(API_PATH.CONNECTIONS, connections, error, loading, {});
  }

  async function createInvitation(alias: string, multiUse: boolean) {
    console.log('> connectionStore.createInvitation');
    error.value = null;
    loading.value = true;

    let invitationData = null;
    // need the await here since the returned invitationData is not one of our stored refs...
    await acapyApi
      .postHttp(
        API_PATH.CONNECTIONS_CREATE_INVITATION,
        {},
        {
          params: { alias, multi_use: multiUse },
        }
      )
      .then((res) => {
        console.log(res);
        invitationData = res.data;
      })
      .then(() => {
        listConnections();
      })
      .catch((err) => {
        error.value = err;
        // console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< connectionStore.createInvitation');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return invitationData;
  }

  async function createOobInvitation(multiUse: boolean, payload = {}) {
    console.log('> connectionStore.createOobInvitation');
    error.value = null;
    loading.value = true;

    let invitationData = null;
    // need the await here since the returned invitationData is not one of our stored refs...
    await acapyApi
      .postHttp(API_PATH.OUT_OF_BAND_CREATE, payload, {
        multi_use: multiUse,
      })
      .then((res) => {
        console.log(res);
        invitationData = res.data;
      })
      .then(() => {
        listConnections();
      })
      .catch((err) => {
        error.value = err;
        // console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< connectionStore.createOobInvitation');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return invitationData;
  }

  async function receiveInvitation(
    invite: string,
    alias: string,
    oob: boolean
  ) {
    console.log('> connectionStore.receiveInvitation');
    error.value = null;
    loading.value = true;

    let acceptedData = null;
    const payload = JSON.parse(invite);

    const url = oob
      ? API_PATH.OUT_OF_BAND_RECIEVE
      : API_PATH.CONNECTIONS_RECEIVE_INVITATION;
    await acapyApi
      .postHttp(url, payload, {
        params: { alias, auto_accept: true },
      })
      .then((res) => {
        console.log(res);
        // don't grab the item, there are other parts of the response data we need (invitation, invitation url)
        acceptedData = res.data;
      })
      .then(() => {
        // do we want to automatically reload? or have the caller of this to load?
        console.log(
          'invitation accepted. the store calls load automatically, but do we want this done "manually"?'
        );
        listConnections();
      })
      .catch((err) => {
        error.value = err;
        // console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< connectionStore.receiveInvitation');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return acceptedData;
  }

  async function deleteConnection(connectionId: string) {
    console.log('> connectionStore.deleteConnection');

    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .deleteHttp(API_PATH.CONNECTION(connectionId))
      .then((res) => {
        result = res.data;
      })
      .then(() => {
        listConnections(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< connectionStore.deleteConnection');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function getConnection(id: string, params: any = {}) {
    loadingItem.value = true;
    return fetchItem(API_PATH.CONNECTIONS, id, error, loadingItem, params);
  }

  async function getInvitation(id: string, params: any = {}) {
    loadingItem.value = true;
    return fetchItem(
      API_PATH.CONNECTIONS_INVITATION(id),
      '',
      error,
      loadingItem,
      params
    );
  }

  // Only going to do alias right now but expand to other params if needed later
  async function updateConnection(id: string, alias: string) {
    console.log('> connectionStore.updateConnection');
    error.value = null;
    loading.value = true;

    let updatedConnection: ConnRecord | null = null;
    const payload: UpdateConnectionRequest = {
      alias,
    };
    await acapyApi
      .putHttp(API_PATH.CONNECTION(id), payload)
      .then((res) => {
        updatedConnection = res.data;
      })
      .then(() => {
        listConnections();
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< connectionStore.updateConnection');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return updatedConnection;
  }

  async function didCreateRequest(did: string, alias: string, myLabel: string) {
    console.log('> connectionStore.didCreateRequest');
    error.value = null;
    loading.value = true;

    await acapyApi
      .postHttp(
        API_PATH.DID_EXCHANGE_CREATE_REQUEST,
        {},
        {
          params: { their_public_did: did, alias, my_label: myLabel },
        }
      )
      .then((res) => {
        console.log(res);
      })
      .then(() => {
        listConnections();
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< connectionStore.didCreateRequest');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
  }

  // private functions
  const connectionLabelValue = (item: any) => {
    let result = null;
    if (item != null) {
      // TODO: determine UX for multiuse blank alias ones
      let alias = 'unknown';
      if (item.alias) {
        alias = item.alias;
      } else if (item.their_label) {
        alias = `Multi-use (${item.their_label})`;
      }
      result = {
        label: alias,
        value: item.connection_id,
        status: item.state,
      };
    }
    return result;
  };

  return {
    connections,
    connectionsDropdown,
    selectedConnection,
    loading,
    loadingItem,
    error,
    filteredConnections,
    filteredInvitations,
    findConnectionName,
    listConnections,
    createInvitation,
    createOobInvitation,
    receiveInvitation,
    deleteConnection,
    getConnection,
    getInvitation,
    updateConnection,
    didCreateRequest,
  };
});

export default {
  useConnectionStore,
};
