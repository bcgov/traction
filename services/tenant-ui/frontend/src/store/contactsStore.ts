import { defineStore } from 'pinia';
import { ref } from 'vue';
import { useTenantApi } from './tenantApi';
import { fetchList } from './utils/fetchList.js';

export const useContactsStore = defineStore('contacts', () => {
  // state
  const contacts: any = ref(null);
  const selectedContact: any = ref(null);
  const loading: any = ref(false);
  const error: any = ref(null);

  // getters

  // actions

  // grab the tenant api
  const tenantApi = useTenantApi();

  async function listContacts() {
    selectedContact.value = null;
    return fetchList('/tenant/v1/contacts/', contacts, error, loading);
  }

  async function createInvitation(alias: string) {
    console.log('> contactsStore.createInvitation');
    error.value = null;
    loading.value = true;

    let invitation_data = null;
    // need the await here since the returned invitation_data is not one of our stored refs...
    await tenantApi
      .postHttp('/tenant/v1/contacts/create-invitation', { alias: alias })
      .then((res) => {
        console.log(res);
        // don't grab the item, there are other parts of the response data we need (invitation, invitation url)
        invitation_data = res.data;
        console.log(invitation_data);
      })
      .then(() => {
        // do we want to automatically reload? or have the caller of this to load?
        console.log('invitation created. the store calls load automatically, but do we want this done "manually"?');
        listContacts();
      })
      .catch((err) => {
        error.value = err;
        //console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< contactsStore.createInvitation');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return invitation_data;
  }

  return { contacts, selectedContact, loading, error, listContacts, createInvitation };
});

export default {
  useContactsStore,
};
