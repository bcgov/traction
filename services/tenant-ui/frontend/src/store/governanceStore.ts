import {
  AddOcaRecordRequest,
  CredDefStorageRecord,
  CredentialDefinitionSendRequest,
  OcaRecord,
  SchemaSendRequest,
} from '@/types/acapyApi/acapyInterface';
import { SchemaStorageRecord } from '@/types';

import { API_PATH } from '@/helpers/constants';
import { defineStore } from 'pinia';
import { computed, Ref, ref } from 'vue';
import { useAcapyApi } from './acapyApi';
import {
  fetchItem,
  fetchList,
  filterByStateActive,
  filterMapSortList,
  sortByLabelAscending,
} from './utils';
import { AddSchemaFromLedgerRequest, StoredSchemaWithCredDefs } from '@/types';

export const useGovernanceStore = defineStore('governance', () => {
  // state
  const storedSchemas: Ref<SchemaStorageRecord[]> = ref([]);
  const selectedSchema: Ref<SchemaStorageRecord | undefined> = ref();

  const storedCredDefs: Ref<CredDefStorageRecord[]> = ref([]);
  const selectedCredentialDefinition: Ref<CredDefStorageRecord | undefined> =
    ref(undefined);

  const ocas: Ref<OcaRecord[]> = ref([]);

  const loading: Ref<boolean> = ref(false);
  const error: any = ref(null);

  // getters

  const schemaList: Ref<StoredSchemaWithCredDefs[]> = computed(() => {
    // For the list of schemas in the schema table, add cred defs
    return storedSchemas.value.map((s: any) => {
      s.credentialDefinitions = storedCredDefs.value.filter(
        (c: CredDefStorageRecord) => c.schema_id === s.schema_id
      );
      return s;
    });
  });

  const schemaLabelValue = (item: any) => {
    let result = null;
    if (item != null) {
      result = {
        label: `${item.name} (${item.version})`,
        value: item.schema_template_id,
        status: item.status,
        can_issue: item.credential_templates.length > 0,
      };
    }
    return result;
  };

  const credDefLabelValue = (item: any) => {
    let result = null;
    if (item != null) {
      result = {
        label: `${item.cred_def_id}`,
        value: item.cred_def_id,
        status: item.state,
      };
    }
    return result;
  };

  const schemaTemplateDropdown = computed(() => {
    return filterMapSortList(
      storedSchemas.value,
      schemaLabelValue,
      sortByLabelAscending,
      filterByStateActive
    );
  });

  const credentialDropdown = computed(() => {
    return filterMapSortList(
      storedCredDefs.value,
      credDefLabelValue,
      sortByLabelAscending
    );
  });

  // actions

  const acapyApi = useAcapyApi();

  async function listStoredSchemas() {
    storedSchemas.value = [];
    return fetchList(API_PATH.SCHEMA_STORAGE, storedSchemas, error, loading);
  }

  async function getStoredSchemas(): Promise<
    SchemaStorageRecord[] | undefined
  > {
    try {
      return (await acapyApi.getHttp(API_PATH.SCHEMA_STORAGE)).data.results;
    } catch (err) {
      console.error(err);
    }
    return;
  }

  async function getStoredCredDefs(): Promise<
    CredDefStorageRecord[] | undefined
  > {
    try {
      return (await acapyApi.getHttp(API_PATH.CREDENTIAL_DEFINITION_STORAGE))
        .data.results;
    } catch (err) {
      console.error(err);
    }
    return;
  }

  async function listStoredCredentialDefinitions() {
    storedCredDefs.value = [];
    return fetchList(
      API_PATH.CREDENTIAL_DEFINITION_STORAGE,
      storedCredDefs,
      error,
      loading
    );
  }

  async function listOcas() {
    ocas.value = [];
    return fetchList(API_PATH.OCAS, ocas, error, loading);
  }

  async function createSchema(payload: SchemaSendRequest) {
    console.log('> governanceStore.createSchema');
    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .postHttp(API_PATH.SCHEMAS, payload)
      .then((res) => {
        result = res.data;
        console.log(result);
      })
      .then(() => {
        // Refresh the schema list
        listStoredSchemas();
      })
      .catch((err) => {
        console.log(err);
        error.value = err;
        // console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< governanceStore.createSchema');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function addSchemaFromLedgerToStorage(
    payload: AddSchemaFromLedgerRequest
  ) {
    console.log('> governanceStore.copySchema');
    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .postHttp(API_PATH.SCHEMA_STORAGE, payload)
      .then((res) => {
        result = res.data;
        console.log(result);
      })
      .then(() => {
        listStoredSchemas();
      })
      .catch((err) => {
        error.value = err;
        // console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< governanceStore.copySchema');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function createCredentialDefinition(
    payload: CredentialDefinitionSendRequest
  ) {
    console.log('> governanceStore.createCredentialDefinition');
    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .postHttp(API_PATH.CREDENTIAL_DEFINITIONS, payload)
      .then((res) => {
        result = res.data;
        console.log(result);
      })
      .then(() => {
        // Refresh the schema list
        listStoredCredentialDefinitions();
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< governanceStore.createCredentialDefinition');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  // async function getSchemaTemplate(id: string, params: any = {}) {
  //   const getloading: any = ref(false);
  //   if (!('credential_templates' in params)) {
  //     // if we do not explicitly say not to get cred templates, get 'em.
  //     params.credential_templates = true;
  //   }
  //   return fetchItem(
  //     API_PATH.GOVERNANCE_SCHEMA_TEMPLATES,
  //     id,
  //     error,
  //     getloading,
  //     params
  //   );
  // }

  async function getCredentialTemplate(id: string, params: any = {}) {
    const getloading: any = ref(false);
    return fetchItem(
      API_PATH.CREDENTIAL_DEFINITIONS,
      id,
      error,
      getloading,
      params
    );
  }

  async function deleteSchema(schemaId: string) {
    console.log('> governanceStore.deleteSchema');
    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .deleteHttp(API_PATH.SCHEMA_STORAGE_ITEM(schemaId), {})
      .then((res) => {
        result = res.data.item;
      })
      .then(() => {
        console.log('schema deleted.');
        listStoredSchemas(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< governanceStore.deleteSchema');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function deleteStoredCredentialDefinition(credDefId: string) {
    console.log('> governanceStore.deleteStoredCredentialDefinition');
    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .deleteHttp(API_PATH.CREDENTIAL_DEFINITION_STORAGE_ITEM(credDefId), {})
      .then((res) => {
        result = res.data.item;
      })
      .then(() => {
        console.log('stored cred deleted.');
        listStoredCredentialDefinitions(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< governanceStore.deleteStoredCredentialDefinition');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function getOca(id: string, params: any = {}) {
    const getloading: any = ref(false);
    return fetchItem(API_PATH.OCAS, id, error, getloading, params);
  }

  async function createOca(payload: AddOcaRecordRequest) {
    console.log('> governanceStore.createOca');
    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .postHttp(API_PATH.OCAS, payload)
      .then((res) => {
        result = res.data.item;
        console.log(result);
        listOcas(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
        // console.log(error.value);
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< governanceStore.createOca');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function deleteOca(ocaId: string) {
    console.log('> governanceStore.deleteOca');
    error.value = null;
    loading.value = true;

    let result = null;

    await acapyApi
      .deleteHttp(API_PATH.OCA(ocaId), {})
      .then((res) => {
        result = res.data.item;
      })
      .then(() => {
        console.log('oca record deleted.');
        listOcas(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    console.log('< governanceStore.deleteOca');

    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  const setSelectedSchemaById = async (id: string) => {
    await getStoredSchemas();
    selectedSchema.value = storedSchemas.value.find(
      (s: SchemaStorageRecord) => s.schema_id === id
    );
  };

  return {
    credentialDropdown,
    error,
    loading,
    ocas,
    schemaList,
    schemaTemplateDropdown,
    selectedCredentialDefinition,
    selectedSchema,
    storedCredDefs,
    storedSchemas,
    // getSchemaTemplate,
    addSchemaFromLedgerToStorage,
    createCredentialDefinition,
    createOca,
    createSchema,
    deleteOca,
    deleteSchema,
    deleteStoredCredentialDefinition,
    getStoredCredDefs,
    getCredentialTemplate,
    getOca,
    getStoredSchemas,
    listOcas,
    listStoredCredentialDefinitions,
    listStoredSchemas,
    setSelectedSchemaById,
  };
});

export default {
  useGovernanceStore,
};
