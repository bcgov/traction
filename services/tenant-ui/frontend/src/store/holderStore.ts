import { defineStore } from 'pinia';
import { ref } from 'vue';
import { fetchItem } from './utils/fetchItem';
import { fetchList } from './utils/fetchList.js';
import { useTenantApi } from './tenantApi';

export const useHolderStore = defineStore('holder', () => {
  // state
  const credentials: any = ref(null);
  const selectedCredential: any = ref(null);
  const presentations: any = ref(null);
  const selectedPresentation: any = ref(null);
  const loading: any = ref(false);
  const error: any = ref(null);

  // getters

  // actions

  async function listCredentials() {
    selectedCredential.value = null;
    return fetchList(
      '/tenant/v1/holder/credentials/',
      credentials,
      error,
      loading
    );
  }

  async function listPresentations() {
    selectedPresentation.value = null;
    return fetchList(
      '/tenant/v1/holder/presentations/',
      presentations,
      error,
      loading
    );
  }

  async function getCredential(id: string, params: any = {}) {
    const getloading: any = ref(false);
    return fetchItem(
      '/tenant/v1/holder/credentials/',
      id,
      error,
      getloading,
      params
    );
  }

  async function getPresentation(id: string, params: any = {}) {
    const getloading: any = ref(false);
    return fetchItem(
      '/tenant/v1/holder/presentations/',
      id,
      error,
      getloading,
      params
    );
  }

  const tenantApi = useTenantApi();

  async function acceptCredentialOffer(credId: string) {
    console.log('> holderStore.acceptCredentialOffer');

    error.value = null;
    loading.value = true;

    let result = null;

    await tenantApi
      .postHttp(`/tenant/v1/holder/credentials/${credId}/accept-offer`, {
        holder_credential_id: credId,
      })
      .then((res) => {
        result = res.data.item;
      })
      .then(() => {
        console.log('credential offer accepted.');
        listCredentials(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function rejectCredentialOffer(credId: string) {
    console.log('> holderStore.rejectCredentialOffer');

    error.value = null;
    loading.value = true;

    let result = null;

    await tenantApi
      .postHttp(`/tenant/v1/holder/credentials/${credId}/reject-offer`, {
        holder_credential_id: credId,
      })
      .then((res) => {
        result = res.data.item;
      })
      .then(() => {
        console.log('credential offer rejected.');
        listCredentials(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  async function deleteHolderCredential(credId: string) {
    console.log('> holderStore.deleteHolderCredential');

    error.value = null;
    loading.value = true;

    let result = null;

    await tenantApi
      .deleteHttp(`/tenant/v1/holder/credentials/${credId}`)
      .then((res) => {
        result = res.data.item;
      })
      .then(() => {
        console.log('credential deleted.');
        listCredentials(); // Refresh table
      })
      .catch((err) => {
        error.value = err;
      })
      .finally(() => {
        loading.value = false;
      });
    if (error.value != null) {
      // throw error so $onAction.onError listeners can add their own handler
      throw error.value;
    }
    // return data so $onAction.after listeners can add their own handler
    return result;
  }

  return {
    credentials,
    presentations,
    selectedCredential,
    selectedPresentation,
    loading,
    error,
    listCredentials,
    listPresentations,
    getCredential,
    getPresentation,
    acceptCredentialOffer,
    rejectCredentialOffer,
    deleteHolderCredential,
  };
});

export default {
  useHolderStore,
};
