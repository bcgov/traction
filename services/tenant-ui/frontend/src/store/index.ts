export { useConfigStore } from './configStore';
export { useContactsStore } from './contactsStore';
export { useGovernanceStore } from './governanceStore';
export { useHolderStore } from './holderStore';
export { useIssuerStore } from './issuerStore';
export { useTenantStore } from './tenantStore';
export { useTokenStore } from './tokenStore';
export { useVerifierStore } from './verifierStore';
