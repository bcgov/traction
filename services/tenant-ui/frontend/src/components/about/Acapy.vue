<template>
  <div class="grid align-items-start">
    <div class="col-12 md:col-6">
      <strong>{{ $t('about.acaPy.info') }}</strong>
      <p class="mt-0">
        {{
          $t('about.acaPy.acapyVersion', {
            version: config.frontend.ariesDetails.acapyVersion,
          })
        }}
      </p>
    </div>
    <div class="col-12 md:col-6 flex justify-content-end pt-0">
      <img src="/img/logo/aries-logo-color.png" class="logo-acapy" />
    </div>
  </div>

  <div class="grid">
    <div class="col-12 md:col-6 lg:col-4">
      <PluginList />
    </div>
  </div>
</template>

<script setup lang="ts">
import { storeToRefs } from 'pinia';
import { useConfigStore } from '@/store/configStore';
// PrimeVue
import PluginList from './PluginList.vue';

const { config } = storeToRefs(useConfigStore());
</script>

<style scoped>
.logo-acapy {
  width: 14em;
}
</style>
