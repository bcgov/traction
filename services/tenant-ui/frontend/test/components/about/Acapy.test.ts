import { useConfigStore } from '@/store/configStore';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import PrimeVue from 'primevue/config';
import { describe, expect, test } from 'vitest';

import Acapy from '@/components/about/Acapy.vue';

const createWrapper = () =>
  mount(Acapy, {
    global: {
      plugins: [PrimeVue, createTestingPinia()],
    },
  });

describe('Acapy', async () => {
  test('mount matches snapshot with expected values', () => {
    expect(createWrapper().html()).toMatchSnapshot();
  });

  test('clicking accordion calls getPluginsList', async () => {
    const store = useConfigStore();
    const wrapper = createWrapper();
    wrapper.getComponent({ name: 'Accordion' }).trigger('click');
  });
});
