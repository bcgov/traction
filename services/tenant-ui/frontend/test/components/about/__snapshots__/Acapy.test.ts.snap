// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Acapy > mount matches snapshot with expected values 1`] = `
"<div data-v-59165757=\\"\\" class=\\"grid align-items-start\\">
  <div data-v-59165757=\\"\\" class=\\"col-12 md:col-6\\"><strong data-v-59165757=\\"\\">about.acaPy.info</strong>
    <p data-v-59165757=\\"\\" class=\\"mt-0\\">about.acaPy.acapyVersion</p>
  </div>
  <div data-v-59165757=\\"\\" class=\\"col-12 md:col-6 flex justify-content-end pt-0\\"><img data-v-59165757=\\"\\" src=\\"/img/logo/aries-logo-color.png\\" class=\\"logo-acapy\\"></div>
</div>
<div data-v-59165757=\\"\\" class=\\"grid\\">
  <div data-v-59165757=\\"\\" class=\\"col-12 md:col-6 lg:col-4\\">
    <div data-v-680038a3=\\"\\" data-v-59165757=\\"\\" class=\\"p-accordion p-component\\" data-pc-name=\\"accordion\\" data-pc-section=\\"root\\">
      <div class=\\"p-accordion-tab\\" data-pc-section=\\"root\\" data-pc-name=\\"accordiontab\\" data-pc-index=\\"0\\" data-p-active=\\"false\\">
        <div class=\\"p-accordion-header\\" data-pc-section=\\"header\\" data-p-highlight=\\"false\\"><a id=\\"undefined_0_header_action\\" class=\\"p-accordion-header-link p-accordion-header-action\\" tabindex=\\"0\\" role=\\"button\\" aria-expanded=\\"false\\" aria-controls=\\"undefined_0_content\\" data-pc-section=\\"headeraction\\"><svg width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"p-icon p-accordion-toggle-icon\\" aria-hidden=\\"true\\" data-pc-section=\\"headericon\\">
              <path d=\\"M4.38708 13C4.28408 13.0005 4.18203 12.9804 4.08691 12.9409C3.99178 12.9014 3.9055 12.8433 3.83313 12.7701C3.68634 12.6231 3.60388 12.4238 3.60388 12.2161C3.60388 12.0084 3.68634 11.8091 3.83313 11.6622L8.50507 6.99022L3.83313 2.31827C3.69467 2.16968 3.61928 1.97313 3.62287 1.77005C3.62645 1.56698 3.70872 1.37322 3.85234 1.22959C3.99596 1.08597 4.18972 1.00371 4.3928 1.00012C4.59588 0.996539 4.79242 1.07192 4.94102 1.21039L10.1669 6.43628C10.3137 6.58325 10.3962 6.78249 10.3962 6.99022C10.3962 7.19795 10.3137 7.39718 10.1669 7.54416L4.94102 12.7701C4.86865 12.8433 4.78237 12.9014 4.68724 12.9409C4.59212 12.9804 4.49007 13.0005 4.38708 13Z\\" fill=\\"currentColor\\"></path>
            </svg><span class=\\"p-accordion-header-text\\" data-pc-section=\\"headertitle\\">about.acaPy.plugins</span>
            <!---->
          </a></div>
        <transition-stub name=\\"p-toggleable-content\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" data-pc-section=\\"accordiontab.transition\\">
          <div id=\\"undefined_0_content\\" class=\\"p-toggleable-content\\" role=\\"region\\" aria-labelledby=\\"undefined_0_header_action\\" data-pc-section=\\"toggleablecontent\\" style=\\"display: none;\\">
            <div class=\\"p-accordion-content\\" data-pc-section=\\"content\\">
              <div data-v-680038a3=\\"\\" class=\\"vjs-tree\\">
                <div class=\\"vjs-tree-node\\">
                  <!---->
                  <!---->
                  <div class=\\"vjs-indent\\">
                    <!---->
                  </div>
                  <!----><span><span class=\\"vjs-value vjs-value-null\\">null</span>
                  <!---->
                  <!----></span>
                </div>
              </div>
            </div>
          </div>
        </transition-stub>
      </div>
    </div>
  </div>
</div>"
`;
